examples/web-wagmi/src/profiles/UseUpdateFollowPolicy.tsx

```diff
@@ -0,0 +1,213 @@
+import {
+  Amount,
+  ChargeFollowPolicy,
+  Erc20,
+  FollowPolicyType,
+  getFollowPolicyTypeFromProfileFieldsFragment,
+  NoFeeFollowPolicy,
+  ProfileFieldsFragment,
+  useCurrencies,
+  useUpdateFollowPolicy,
+} from '@lens-protocol/react';
+import { useState } from 'react';
+
+import { UnauthenticatedFallback } from '../components/UnauthenticatedFallback';
+import { WhenLoggedInWithProfile } from '../components/auth/auth';
+import { Loading } from '../components/loading/Loading';
+import { never } from '../utils';
+
+const followPolicyTypeToDescriptionMap: Record<FollowPolicyType, string> = {
+  [FollowPolicyType.ONLY_PROFILE_OWNERS]: 'Only holders of Lens profiles can follow',
+  [FollowPolicyType.ANYONE]: 'Anyone can follow',
+  [FollowPolicyType.NO_ONE]: 'No one can follow',
+  [FollowPolicyType.CHARGE]: 'Anyone can follow, but they must pay a fee',
+};
+
+function resolveFollowPolicy({
+  followPolicyType,
+  amount,
+  recipient,
+}: {
+  followPolicyType: FollowPolicyType;
+  amount?: Amount<Erc20>;
+  recipient?: string;
+}): ChargeFollowPolicy | NoFeeFollowPolicy {
+  if (followPolicyType === FollowPolicyType.CHARGE) {
+    return {
+      type: FollowPolicyType.CHARGE,
+      amount: amount ?? never(),
+      recipient: recipient ?? never(),
+    };
+  }
+
+  return {
+    type: followPolicyType,
+  };
+}
+
+type UpdateButtonTextProps = {
+  isTxPending: boolean;
+  currentFollowModule: FollowPolicyType | null;
+  followPolicyTypeToUpdate: FollowPolicyType | null;
+};
+
+function UpdateButtonText({
+  isTxPending,
+  currentFollowModule,
+  followPolicyTypeToUpdate,
+}: UpdateButtonTextProps) {
+  if (isTxPending) return <>Updating...</>;
+
+  if (currentFollowModule === followPolicyTypeToUpdate) {
+    return <>This is your current follow policy</>;
+  }
+
+  return <>Update</>;
+}
+
+type UpdateFollowPolicyProps = {
+  profile: ProfileFieldsFragment;
+};
+
+function UpdateFollowPolicy({ profile }: UpdateFollowPolicyProps) {
+  const currentFollowModule = getFollowPolicyTypeFromProfileFieldsFragment(profile);
+  const [followPolicyTypeToUpdate, setFollowPolicyTypeToUpdate] = useState<FollowPolicyType | null>(
+    currentFollowModule,
+  );
+
+  const { data: currencies, loading } = useCurrencies();
+  const { updateFollowPolicy, isPending, error } = useUpdateFollowPolicy();
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    if (!followPolicyTypeToUpdate || !currencies) return;
+
+    const formData = new FormData(event.currentTarget);
```

Comment:
Thanks you for highlight this approach in my PR https://github.com/lens-protocol/lens-sdk/pull/77. Why not extend this to `followPolicyTypeToUpdate` as well? It seems something that can also be part of form data.